src/go/log/log_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLevel_String(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		level Level
+		want  string
+	}{
+		{
+			level: DebugLevel,
+			want:  "DEBUG",
+		},
+		{
+			level: InfoLevel,
+			want:  "INFO",
+		},
+		{
+			level: WarnLevel,
+			want:  "WARN",
+		},
+		{
+			level: ErrorLevel,
+			want:  "ERROR",
+		},
+		{
+			level: 100,
+			want:  "INVALID (100)",
+		},
+	}
+
+	for _, test := range tests {
+		have := test.level.String()
```

Comment:
nit / question: I was trained to use `got` not `have`. Do you see preference in open source for have?

Reply:
not sure where I picked up `have`, will go for `got`

Reply:
To confirm this I actually found documentation with this phrasing on the [golang github repo](https://github.com/golang/go/issues/34384).